eng/pipelines/agent-software-inventory.yml

```diff
@@ -7,183 +7,92 @@ pr:
-      } > programfiles-inventory.csv
-    displayName: (Windows) Program Files
-    workingDirectory: $(Build.ArtifactStagingDirectory)
-    condition: and(succeeded(), eq(variables.Host, 'Windows'))
-
-  - pwsh: |
-      Get-ChildItem "${env:ProgramFiles(x86)}" | ForEach-Object {
-        "$(Host), $(OSVmImage), ProgramFilesX86, $($_.Name), , all"
-      } >> programfiles-inventory.csv
-    displayName: (Windows) Program Files (x86)
-    workingDirectory: $(Build.ArtifactStagingDirectory)
-    condition: and(succeeded(), eq(variables.Host, 'Windows'))
-
-  # - pwsh: |
-  #     Get-ChildItem env: | ForEach-Object {
-  #       "$(Host), $(OSVmImage), Environment, $($_.Name), $($_.Value), all"
-  #     } >> env-inventory.csv
-  #   displayName: Environment Variables
-  #   workingDirectory: $(Build.ArtifactStagingDirectory)
-
-  - pwsh: |
-      $root = $env:VCPKG_INSTALLATION_ROOT
-      if (!$root) {
-        $root = 'C:\vcpkg'
-      }
-      Get-ChildItem "$root\packages" -ErrorAction Ignore | ForEach-Object {
-        "$(Host), $(OSVmImage), Environment, $($_.Name), $($_.Value), all"
-      } >> vcpkg-inventory.csv
-    displayName: (Windows) Vcpkg Packages
-    workingDirectory: $(Build.ArtifactStagingDirectory)
-    condition: and(succeeded(), eq(variables.Host, 'Windows'))
-
-  - publish: $(Build.ArtifactStagingDirectory)
-    displayName: Publish inventory-$(OSVmImage)
-    artifact: inventory-$(OSVmImage)
-
-- job: assemble
-  dependsOn: inventory
-  condition: succeededOrFailed()
-  pool:
-    vmImage: ubuntu-latest
-  steps:
-  - checkout: none
-
-  - download: current
-    displayName: Download artifacts
-    patterns: |
-      **/*.csv
-
-  - pwsh: >
-      Get-ChildItem -Filter *.csv -Recurse
-      | Get-Content
-      | ConvertFrom-Csv -Header Host, VmImage, Type, Name, Version, Architecture
-      | Sort-Object Host, VmImage, Type, Name, Version, Architecture
-      | ConvertTo-Csv
-      > inventory.csv
-    displayName: Combine and sort inventory
-    workingDirectory: $(Pipeline.Workspace)
-
-  - publish: $(Pipeline.Workspace)/inventory.csv
-    artifact: inventory
+  - job: inventory_azure_pipelines
+    strategy:
+      matrix:
+        Linux:
+          Pool: Azure Pipelines
+          OSVmImage: ubuntu-latest
+          Host: Ubuntu
+        Windows:
+          Pool: Azure Pipelines
+          OSVmImage: windows-latest
+          Host: Windows
+        MacOS:
+          Pool: Azure Pipelines
+          OSVmImage: macOS-latest
+          Host: MacOS
+
+
+    pool:
+      name: $(Pool)
+      vmImage: $(OSVmImage)
+
+    steps:
+    - template: /eng/pipelines/templates/steps/inventory-steps.yml
+
+  - job: inventory_custom_agents_linux
+    strategy:
+      matrix:
+        CustomLinux22:
+          Pool: azsdk-pool
+          OSVmImage: ubuntu-22.04
+          Demands: ImageOverride -equals ubuntu-22.04
+          Host: Ubuntu
+        CustomLinux24:
+          Pool: azsdk-pool
+          OSVmImage: ubuntu-24.04
```

Comment:
I guess given the nature of this pipeline you are just hardcoding instead of using the image variables. I'm fine with that.